src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -182,8 +183,9 @@ public function getRedirectResponse()
             throw new RuntimeException('This response does not support redirection.');
         }
 
+        /** @var $this RedirectResponseInterface */
         if ('GET' === $this->getRedirectMethod()) {
-            return HttpRedirectResponse::create($this->getRedirectUrl());
+            return new RedirectResponse($this->getRedirectUrl());
```

Comment:
I thought we can inject $request and $response. So I am not  really thinking about `ServerRequest` factory. I will try to go through the code when I get some time.

Thank you for the help.